Source/WebCore/platform/graphics/gstreamer/eme/CDMThunder.cpp

```diff
@@ -440,6 +446,20 @@ void CDMInstanceSessionThunder::keyUpdatedCallback(KeyIDType&& keyID)
     auto keyStatus = status(keyID);
     GST_DEBUG("updated with with key status %s", toString(keyStatus));
     m_doesKeyStoreNeedMerging |= m_keyStore.add(KeyHandle::create(keyStatus, WTFMove(keyID), BoxPtr<OpenCDMSession>(m_session)));
+    auto instance = cdmInstanceThunder();
+    if (instance && GStreamerEMEUtilities::isPlayReadyKeySystem(instance->keySystem())) {
+        // PlayReady corner case hack: It happens that the key ID
```

Comment:
Can this hack be conditional to `isThunderRanked()` please?

Reply:
I don't think it is a good idea because you could still have thunder running but not necessarily upranked. Anyway, I think conditional compilation for `THUNDER` would be a good idea.

Reply:
And it is already guarded as this is the _CDMThunder.cpp_ file.

Reply:
> Can this hack be conditional to `isThunderRanked()` please?

Actually this function doesn't make much sense anymore since the ClearKey-specific decryptor was removed. If you want to handle that encryption format you'd need to implement a Thunder thing for it. No need to play with element ranks :)

Reply:
There's no code yet for detecting a sparkle-cdm runtime, so you can keep this hack un-guarded. Probably will need a revisit in case a sparkle-cdm playready plugin appears...
Would be nice though to clean out the `isThunderRanked()` thing, can you do it?

Reply:
Can we do it in a follow up?